### PR TITLE
fix: [ANDROAPP-6493] take incident date into account for scheduling events

### PR DESCRIPTION
**Pull request by xavimolloy**
## Description
Please include a summary of the change and include the related jira issue if it exists.

[ jira issue ](https://dhis2.atlassian.net/browse/ANDROAPP-

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [ ] 11.X - 13.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

**Comment by ferdyrod** [comment]
Could it be a good a idea to create a repository class with all the new d2.[xxx]module() calls? Also, this is a great opportunity to move the usecase to the tracker module. What do you think?

**Review by Balcan — CHANGES_REQUESTED** [review]
Can you add some tests for the CreateEventUseCase class? It would be also be interesting adding new tests to check the period date generation.

**Review by ferdyrod — APPROVED** [review]

File: tracker/src/main/kotlin/org/dhis2/tracker/events/CreateEventUseCaseRepository.kt [hunk 1]
```diff
@@ -1,8 +1,6 @@
-package org.dhis2.usescases.programEventDetail.usecase
+package org.dhis2.tracker.events
```
Comment (andresmr) [inline tracker/src/main/kotlin/org/dhis2/tracker/events/CreateEventUseCaseRepository.kt:1]: very good!

File: form/src/main/java/org/dhis2/form/data/EventRepository.kt [hunk 2]
```diff
@@ -426,21 +427,31 @@ class EventRepository(
...
-                    maxDate = dateUtils.today,
+                    maxDate = getStartOfDay(Date()),
                 )
             } else {
                 null
             }
         }
     }
 
+    private fun getStartOfDay(date: Date): Date {
```
Comment (andresmr) [inline form/src/main/java/org/dhis2/form/data/EventRepository.kt:438]: I see that you have created this three times, what do you think about create it once in DateUtils and use it from there?

File: tracker/src/main/kotlin/org/dhis2/tracker/events/CreateEventUseCaseRepository.kt [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                val minDaysFromStart = getMinDaysFromStartByProgramStage(programStage)
+                calendar.add(DAY_OF_YEAR, minDaysFromStart)
+            } else {
+                calendar.add(DAY_OF_YEAR, programStage?.standardInterval() ?: 0)
+            }
+            if (programStage?.periodType() == null) {
+                eventRepository.setEventDate(calendar.time ?: dateUtils.today)
+            } else {
+                val nextAvailablePeriod = dateUtils.getNextPeriod(programStage.periodType(), calendar.time ?: dateUtils.today, 0)
+                eventRepository.setEventDate(nextAvailablePeriod)
```
Comment (andresmr) [inline tracker/src/main/kotlin/org/dhis2/tracker/events/CreateEventUseCaseRepository.kt:47]: to improve readability, can you extract all the login regarding getting the event date to a method and then here just `eventRepository.setEventDate(getEventDate(...))` or something like, what do you think?

Comment (xavimolloy) [reply tracker/src/main/kotlin/org/dhis2/tracker/events/CreateEventUseCaseRepository.kt]: of course! I just updated the PR

**Review by andresmr — APPROVED** [review]

**Review by Balcan — APPROVED** [review]

**Review by Balcan — APPROVED** [review]

**Pull request merged**